### ISSUE TITLE
Integral result incorrect: simplifies sign away

### ISSUE THREAD

**Issue by cidermole** [sympy/sympy#2975]
Here's an integral, a part of an expression for an electric field: `integrate (1/(y**2+C)**(3/2), y)`

The expected answer is `y/(C * sqrt(C+y**2))`, as confirmed by Wolfram Alpha:
http://www.wolframalpha.com/input/?i=integrate+1%2F%28y**2%2BC%29**%283%2F2%29+dy

The answer from SymPy 0.7.5-1 in IPython2 is:

```
In [1]: from sympy import *
In [2]: from __future__ import division
In [3]: y, C, w = var('y C w')

In [4]: integrate(1/(y**2+C)**(3/2), y)
Out[4]: 0.564189583547911*sqrt(pi)*1/C/sqrt(C/y**2 + 1)
```

This looks equal to the correct answer, but it is not. Considering the integral range [-w/2:w/2], SymPy yields a 0 result, which is obviously incorrect:

```
In [5]: integrate(1/(y**2+C)**(3/2), (y, -w/2, w/2))
Out[5]: 0
```

The expected, correct answer would be w/(C \* sqrt(C+w**2/4)).

**Comment by smichr** [comment]
Integrating across 0 is perhaps the problem? The indefinate looks ok:

```python
>>> i=integrate (1/(y**2+C)**(S(3)/2), y);i
1/(C*sqrt(C/y**2 + 1))
>>> together(i)
1/(C*sqrt((C + y**2)/y**2))
>>> pprint(_.subs(y,Dummy('y',positive=True)))
      y
-------------
     ________
    /  2
C*\/  y  + C
```

**Comment by jcrist** [comment]
Still valid in master. Floating point exponent gives same result as rational (what @smichr wrote above), just not simplified. The definite integral is the culprit. Looking further, without a positive assumption for `y`, the numerator is in an `abs`, which is why the definite integral evaluates to 0. This could be an assumptions problem.

**Comment by oscarbenjamin** [comment]
This seems to give the expected answers now:
```julia
In [1]: C = Symbol('C')

In [2]: integrate (1/(y**2+C)**(3/2), y)
Out[2]: 
                      -1.5  
0.564189583547756⋅√π⋅C    ⋅y
────────────────────────────
            ________        
           ╱      2         
          ╱      y          
         ╱   1 + ──         
       ╲╱        C          

In [3]: integrate (1/(y**2+C)**(S(3)/2), y)
Out[3]: 
        y         
──────────────────
          ________
         ╱      2 
 3/2    ╱      y  
C   ⋅  ╱   1 + ── 
     ╲╱        C  
In [5]: w = Symbol('w')

In [6]: integrate(1/(y**2+C)**(3/2), (y, -w/2, w/2))
Out[6]: 
                      -1.5  
0.564189583547756⋅√π⋅C    ⋅w
────────────────────────────
            _________       
           ╱       2        
          ╱       w         
         ╱   1 + ───        
       ╲╱        4⋅C        

In [7]: integrate(1/(y**2+C)**(S(3)/2), (y, -w/2, w/2))
Out[7]: 
         w         
───────────────────
          _________
         ╱       2 
 3/2    ╱       w  
C   ⋅  ╱   1 + ─── 
     ╲╱        4⋅C 
```

This can be closed if a test is added.

**Comment by Smit-create** [comment]
@oscarbenjamin I think this issue should be closed as the test case has been added in #18220.

**Comment by oscarbenjamin** [comment]
Yeas it should be closed. Thanks for pointing that out.
